status/views.py

```diff
@@ -1,97 +1,109 @@
-            data[0]['details'] = "PING failed"
-    except Exception as e:
-        data[0]['status'] = 'offline'
-        data[0]['details'] = "PING failed"
-
-
-    # Test postgresql connection
-    try:
-        with connection.cursor() as cursor:
-            cursor.execute("SELECT 1");
-        data[1]['status'] = 'online'
-    except:
-        data[1]['status'] = 'offline'
-        data[1]['details'] = 'Could not execute query `SELECT 1\''
-
-    try:
-        response = requests.get(f"{settings.CHALLENGE_SERVER_URL}/status",
-                            headers={"Authorization": settings.CHALLENGE_SERVER_API_KEY})
-        if response.status_code == 200:
-            sysinfo = response.json()
-            reasons = []
-            if sysinfo['Loads'][0] > 90:
-                reasons.append(f"High CPU load: {sysinfo['Loads'][0]}")
-            if sysinfo['FreeRam'] / sysinfo['TotalRam'] < 0.2:
-                reasons.append(f"High RAM usage: {sysinfo['FreeRam'] / 1024}MB free out of {sysinfo['TotalRam']}")
-            if len(reasons) != 0:
-                data[2]['status'] = 'partial'
-                data[2]['details'] = '\n'.join(reasons)
+
+class StatusView(APIView):
+    permission_classes = (IsAdminUser,)
+
+    def get(self, request):
+        connection_info = []
+
+        redis_status, redis_details = self.test_redis_connection()
+        connection_info.append({"name": "Redis", "status": redis_status, "details": redis_details})
+
+        postgres_status, postgres_details = self.test_postgresql_connection()
+        connection_info.append({"name": "PostgreSQL", "status": postgres_status, "details": postgres_details})
+
+        andromeda_status, andromeda_details = self.get_andromeda_status()
+        connection_info.append({"name": "Andomeda", "status": andromeda_status, "details": andromeda_details})
+
+        if settings.SEND_MAIL:
+            mailusv_status, mailusv_details = self.test_mailusv_connection()
+            connection_info.append({"name": "Mail Daemon", "status": mailusv_status, "details": mailusv_details})
+
+        return FormattedResponse(connection_info)
+
+    def test_redis_connection(self):
+        status, details = 'unknown', ''
+
+        try:
+            rs_details = settings.CONFIG['REDIS']
+            rs = redis.Redis(host=rs_details['HOST'], port=rs_details['PORT'])
+
+            if rs.ping() == True:
+                status = 'online'
             else:
-                data[2]['status'] = 'online'
-        else:
-            data[2]['status'] = 'unknown'
-            data[2]['details'] = 'Error connecting to /status'
-    except Exception as e:
-        data[2]['status'] = 'offline'
-        data[2]['details'] = str(e)
-
-    if settings.SEND_MAIL:
-        data.append(
-        {
-            "name": "Mail Daemon",
-            "status": "unknown",
-            "details": ''
-        })
+                status = 'offline'
+                details = "PING failed"
```

Comment:
For better compatibility with frontend, these would be better as i18n strings such as redis_ping_failed